### PR TITLE
Add a global config for col capitalization

### PR DESCRIPTION
**Pull request by utkarsharma2**
# Description
## What is the current behavior?
Users expressed confusion on why the column names for output tables are lowercased. It's due to this [default setting](https://github.com/astronomer/astro-sdk/blob/main/python-sdk/src/astro/sql/operators/dataframe.py#L110).

closes: #917

## What is the new behavior?
The default behavior should be to not change casing and it should be configureable with a global variable.

That way we can avoid hardcoding this setting into the Astro Build dagfile generator script.

## Does this introduce a breaking change?
Nope

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: python-sdk/src/astro/constants.py [hunk 1]
```diff
@@ -65,3 +65,4 @@ def __str__(self) -> str:
 MergeConflictStrategy = Literal["ignore", "update", "exception"]
 
 ColumnCapitalization = Literal["upper", "lower", "original"]
+DEFAULT_COLUMN_CAPITALIZATION = "original"
```
Comment (kaxil) [inline python-sdk/src/astro/constants.py:68]:
```suggestion
```

Comment (kaxil) [reply python-sdk/src/astro/constants.py:68]: We can just add this as fallback in settings.py

File: python-sdk/src/astro/settings.py [hunk 2]
```diff
@@ -41,3 +41,8 @@
 AUTO_ADD_INLETS_OUTLETS = conf.getboolean(
     "astro_sdk", "auto_add_inlets_outlets", fallback=True
 )
+
+
+COLUMN_CAPITALIZATION = conf.get(
+    "astro_sdk", "column_capitalization", fallback=DEFAULT_COLUMN_CAPITALIZATION
+)
```
Comment (kaxil) [inline python-sdk/src/astro/settings.py:48]:
```suggestion
COLUMN_CAPITALIZATION = conf.get("astro_sdk", "column_capitalization", fallback="original")
```

File: python-sdk/src/astro/settings.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 from airflow.configuration import conf
-from astro.constants import DEFAULT_SCHEMA
+from astro.constants import DEFAULT_COLUMN_CAPITALIZATION, DEFAULT_SCHEMA
```
Comment (kaxil) [inline python-sdk/src/astro/settings.py:2]:
```suggestion
from astro.constants import DEFAULT_SCHEMA
```

File: python-sdk/src/astro/settings.py [hunk 4]
```diff
@@ -41,3 +41,8 @@
 AUTO_ADD_INLETS_OUTLETS = conf.getboolean(
     "astro_sdk", "auto_add_inlets_outlets", fallback=True
 )
+
+
+COLUMN_CAPITALIZATION = conf.get(
+    "astro_sdk", "column_capitalization", fallback="original"
+)
```
Comment (kaxil) [inline python-sdk/src/astro/settings.py:48]: We should add docs about it in https://github.com/astronomer/astro-sdk/blob/main/python-sdk/docs/configurations.rst

Comment (kaxil) [reply python-sdk/src/astro/settings.py:48]: Also, you have the following in the PR description

>Does this introduce a breaking change?
> Nope

But this does introduce a breaking change. When writing the release notes, we should ask users to change the config if needed if they relied on this behavior and mention that we consider it a "bug" hence the change

Comment (feluelle) [reply python-sdk/src/astro/settings.py:48]: Why is that a breaking change? The default does not change as we are setting `COLUMN_CAPITALIZATION` default/fallback to `original`.

File: python-sdk/src/astro/files/types/csv.py [hunk 5]
```diff
@@ -12,7 +13,10 @@ class CSVFileType(FileType):
     """Concrete implementation to handle CSV file type"""
 
     def export_to_dataframe(
-        self, stream, columns_names_capitalization="original", **kwargs
+        self,
+        stream,
+        columns_names_capitalization=settings.COLUMN_CAPITALIZATION,
+        **kwargs,
```
Comment (dimberman) [inline python-sdk/src/astro/files/types/csv.py:19]: Can you add type hints here

File: python-sdk/src/astro/files/types/ndjson.py [hunk 6]
```diff
@@ -14,7 +15,10 @@ class NDJSONFileType(FileType):
     """Concrete implementation to handle NDJSON file type"""
 
     def export_to_dataframe(
-        self, stream, columns_names_capitalization="original", **kwargs
+        self,
+        stream,
+        columns_names_capitalization=settings.COLUMN_CAPITALIZATION,
+        **kwargs,
```
Comment (dimberman) [inline python-sdk/src/astro/files/types/ndjson.py:21]: type hints here as well

File: python-sdk/src/astro/settings.py (hunk 4 above)
Comment (dimberman) [reply python-sdk/src/astro/settings.py:48]: +1

File: python-sdk/tests/sql/operators/test_dataframe.py [hunk 7]
```diff
@@ -334,3 +335,24 @@ def sample_df_1(**kwargs):
 
     task = sample_df_1(output_table=output_table)
     assert task.operator.outlets == []
+
+
+def test_col_capitalization_default_dataframe(sample_dag):
+    """Test that capitalization of col remains the same"""
+
+    assert settings.COLUMN_CAPITALIZATION == "original"
```
Comment (dimberman) [inline python-sdk/tests/sql/operators/test_dataframe.py:343]: You can use pytest parameters to make this test apply to all three known settings (and maybe even an invalid setting?)

**Review by dimberman — CHANGES_REQUESTED** [review]

File: python-sdk/src/astro/settings.py (hunk 4 above)
Comment (feluelle) [reply python-sdk/src/astro/settings.py:48]: Thanks @kaxil for pointing this out 👍

**Comment by utkarsharma2** [comment]
We no longer need this PR, since we have added a fix in a separate PR.

**Pull request closed without merging**